container/go/pkg/compat/config.go

```diff
@@ -42,6 +42,29 @@ const (
 	defaultTimestamp = "1970-01-01T00:00:00Z"
 )
 
+var (
+	validHealcheckTypes = map[string]func([]string) error{
+		"NONE": func(args []string) error {
+			if len(args) == 1 {
+				return nil
+			}
+			return errors.New("NONE doesn't accept any extra args")
```

Comment:
I'm not super familiar with Go, but I think it's common practice to put the error case in the indented if statement and have the default be the positive best case path.

Reply:
Yeah. I don't have a strong opinion on that so I've moved the error case under a conditional branch as you offered